swagger-contract-tests/src/test/resources/contract/swagger2/declaration-spring-data-rest.json

```diff
@@ -223,7 +223,7 @@
           "200": {
             "description": "OK",
             "schema": {
-              "$ref": "#/definitions/ResourceOfPerson"
+              "$ref": "#/definitions/Resource_1OfPerson_1"
```

Comment:
@MaksimOrlov Im getting to merging this PR finally. 

However, this doesn't look right to me... There shouldn't be 10 different `Resources`. We shouldn't have n different versions of Resource when the model for sure doesn't ever change and is only used in return values.

I would expect `Resource_2OfAddress_1`, `Resource_1OAddress_1` so on and so forth

Reply:
@dilipkrish, there is no 10 different `Resources`

So, we have here:

`Resource_1OfPerson_1`
`Resource_2OfAddress_1`
`Resource_3OfCategory_1`
`Resource_4OfTag_1`

`Resources_1OfListOfPerson_1`
`Resources_2OfTag_1`
`Resources_3OfPerson_1`
`Resources_4OfAddress_1`
`Resources_5OfCategory_1`

```
public class Resource<T> extends ResourceSupport {

	private final T content;

        @JsonUnwrapped
	@XmlAnyElement
	public T getContent() {
		return content;
	}
}
```

So, at the runtime T can be any model, and **model** `Resource` has a `ModelRef` inside, that points to any **model**. We compare models at the runtime, SpringFox can see that type of the parameters are different, so models are different. Will name it and increase the digit.

Here we have Person_1 and Person_2, Address_1 and Address_2 and so on, because of ceclic links.

Reply:
Yeah I see that, Im wondering if we can make it better. It will be weird for code generation as well based on these models.

For e.g. `Resources_1OfListOfPerson_1` doesnt have Resource_1**List_1**OfPerson, can we not have Resource also be similar to `List`?

Reply:
`List` is not a model, it is ignored by `ModelProvider` and `TypeNameExtractor`.

I've added `Map` as an exception, see [that line](https://github.com/springfox/springfox/pull/2056/files#diff-4e40ae816770c5033e017cebee12a6acR141). We can do something like that. F.e. add *NotAdjustableTypes*

But here could be conflicts with user defined types with name *Map* or *Resource*.